probe-rs/src/rtt/channel.rs

```diff
@@ -16,14 +16,172 @@ pub trait RttChannel {
     fn buffer_size(&self) -> usize;
 }
 
+#[repr(C)]
+#[derive(Debug, FromZeroes, FromBytes, Copy, Clone)]
+pub(crate) struct RttChannelBufferInner<T> {
+    header: RttChannelBufferInnerHeader<T>,
+    buffer_offset: RttChannelBufferInnerBufferOffset<T>,
+    flags: T,
+}
+
+impl<T> RttChannelBufferInner<T> {
+    pub fn write_buffer_ptr_offset(&self) -> usize {
+        std::mem::size_of::<RttChannelBufferInnerHeader<T>>()
+    }
+
+    pub fn read_buffer_ptr_offset(&self) -> usize {
+        std::mem::size_of::<RttChannelBufferInnerHeader<T>>() + std::mem::size_of::<T>()
+    }
+
+    pub fn flags_offset(&self) -> usize {
+        std::mem::size_of::<Self>() - std::mem::size_of::<T>()
```

Comment:
Can we use [offset_of!](https://doc.rust-lang.org/stable/std/mem/macro.offset_of.html) here? Also, considering these functions are (almost?) always used to calculate some memory access offset, I think it would be better if all three of these returned `u64`